service/controller.go

```diff
@@ -403,15 +405,46 @@ func (s *service) CreateVolume(
 		_, err = isiConfig.isiSvc.client.GetPolicyByName(ctx, ppName)
 		if err != nil {
 			if apiErr, ok := err.(*isiApi.JSONError); ok && apiErr.StatusCode == 404 {
-				err := isiConfig.isiSvc.client.CreatePolicy(ctx, ppName, rpoint, isiPath+"/"+vgName, isiPath+"/"+vgName, remoteSystemEndpoint)
+				err := isiConfig.isiSvc.client.CreatePolicy(ctx, ppName, rpoint, isiPath+"/"+vgName, isiPath+"/"+vgName, remoteSystemEndpoint, remoteIsiConfig.ReplicationCertificateID, true)
 				if err != nil {
 					return nil, status.Errorf(codes.Internal, "can't create protection policy %s", err.Error())
 				}
+				err = isiConfig.isiSvc.client.WaitForPolicyLastJobState(ctx, ppName, isi.FINISHED)
+				if err != nil {
+					return nil, status.Errorf(codes.Internal, "policy job couldn't reach FINISHED %s", err.Error())
+				}
 			} else {
 				return nil, status.Errorf(codes.Internal, "can't ensure protection policy exists %s", err.Error())
 			}
 		}
 
+		_, err = remoteIsiConfig.isiSvc.client.GetPolicyByName(ctx, ppName)
+		if err != nil {
+			if apiErr, ok := err.(*isiApi.JSONError); ok && apiErr.StatusCode == 404 {
+				err := remoteIsiConfig.isiSvc.client.CreatePolicy(ctx, ppName, rpoint, isiPath+"/"+vgName, isiPath+"/"+vgName, isiConfig.Endpoint, isiConfig.ReplicationCertificateID, true)
+				if err != nil {
+					return nil, status.Errorf(codes.Internal, "can't create protection policy %s", err.Error())
+				}
+				err = remoteIsiConfig.isiSvc.client.WaitForPolicyLastJobState(ctx, ppName, isi.FINISHED)
+				if err != nil {
+					return nil, status.Errorf(codes.Internal, "policy job couldn't reach FINISHED state %s", err.Error())
+				}
+			}
+		}
+
+		err = isiConfig.isiSvc.client.AllowWrites(ctx, ppName)
+		if err != nil {
+			return nil, status.Errorf(codes.Internal, "can't allow writes on local site %s", err.Error())
+		}
+		err = remoteIsiConfig.isiSvc.client.DisablePolicy(ctx, ppName)
+		if err != nil {
+			return nil, status.Errorf(codes.Internal, "can't disable the policy on TGT %s", err.Error())
+		}
+		err = remoteIsiConfig.isiSvc.client.WaitForPolicyEnabledFieldCondition(ctx, ppName, false)
+		if err != nil {
+			return nil, status.Errorf(codes.Internal, "policy couldn't reach disabled condition on TGT %s", err.Error())
```

Comment:
Dont see any logs which can identify the volume for which replication is failing.

Reply:
Can you elaborate a little more? 
Don't we already have volume name in the parameters of REQ CreateVolume? I think it would be pretty easy to identify for what volume it is failing, just take a number of the failed REP and find the corresponding REQ.

Reply:
Okay.